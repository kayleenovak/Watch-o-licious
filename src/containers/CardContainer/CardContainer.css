.card-container {
  max-width: 100vw;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  height: 100%;
}