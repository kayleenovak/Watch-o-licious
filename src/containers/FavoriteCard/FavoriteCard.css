.fav-card {
  display: flex;
  width: 500px;
  position: relative;
  margin: 30px;
}

.fav-card-background {
  position: absolute;
  height: 280px;
  width: 500px;
  top: 40px;
  bottom: 0;
  background: #1f151c;
  border-radius: 10px;
  box-shadow: 2px 2px 20px black;
}

.fav-card-image,
.fav-card-info {
  position: relative;
  z-index: 2000;
}

.episode-information {
  font-family: 'Montserrat', sans-serif;
  margin-top: 28px;
}

.episode-series {
  font-size: 14px;
}

.episode-information,
.episode-series,
.episode-runtime,
.episode-airdate {
  color: white;
}

.episode-series,
.episode-runtime,
.episode-airdate {
  font-family: 'Raleway', sans-serif;
  margin: 10px 0;
}

.fav-card-image {
  margin: 0 0 20px 25px;
  box-shadow: 2px 2px 10px black;
}

.fav-card-info {
  padding: 20px;
}

.episode-information {
  color: white;
}

.view-series-btn {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background: linear-gradient(to bottom, #aea7b9 5%, #74707a 100%);
  background-color: #aea7b9;
  border-radius: 3px;
  border: 2px solid #292929;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial;
  font-size: 14px;
  font-weight: bold;
  padding: 6px 10px;
  text-decoration: none;
  text-shadow: 1px 1px 0px black;
  margin: -5px 15px 0 0px;
}

.view-series-btn:hover {
  background: linear-gradient(to bottom, #74707a 5%, #aea7b9 100%);
  background-color: #74707a;
}

.view-series-btn:active {
  position: relative;
  top: 1px;
}

.fav-card-btns {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background: linear-gradient(to bottom, #aea7b9 5%, #74707a 100%);
  background-color: #aea7b9;
  border-radius: 3px;
  border: 2px solid #292929;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial;
  font-size: 12px;
  font-weight: bold;
  padding: 4px 8px;
  text-decoration: none;
  text-shadow: 1px 1px 0px black;
  margin: -5px 15px 5px 0px;
}

.fav-card-btns:hover {
  background: linear-gradient(to bottom, #74707a 5%, #aea7b9 100%);
  background-color: #74707a;
}

.fav-card-btns:active {
  position: relative;
  top: 1px;
}
