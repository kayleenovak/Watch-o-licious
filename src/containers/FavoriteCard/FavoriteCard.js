import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addToTracked, removeFromTracked, toggleFavorite, toggleWatched, toggleWatchList } from '../../actions/index.js'

export class FavoriteCard extends Component {
  render() {
    const matchedTvShow = this.props.tvShows.find(show => {
      return show.id === parseInt(this.props.episode.showId)
    })
    return (
      <section>
        <img src={matchedTvShow.image} />
        <h3>Episode: {this.props.episode.title}</h3>
        <h5>{matchedTvShow.title}</h5>
        <button>Favorite</button>
        <button>Watch List</button>
        <button>Watched</button>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  tvShows: state.tvShows
})

export const mapDispatchToProps = (dispatch) => ({
  addToTracked: (episode) => dispatch(addToTracked(episode)),
  removeFromTracked: (episode) => dispatch(removeFromTracked(episode)),
  toggleFavorite: (episode) => dispatch(toggleFavorite(episode)),
  toggleWatched: (episode) => dispatch(toggleWatched(episode)),
  toggleWatchList: (episode) => dispatch(toggleWatchList(episode))
})

export default connect(mapStateToProps, mapDispatchToProps)(FavoriteCard)