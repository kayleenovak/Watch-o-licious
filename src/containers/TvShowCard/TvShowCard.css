.tv-show-card {
  width: 275px;
  height: 405px;
}

.tv-card-title {
  width: 210px;
  font-family: 'Montserrat', sans-serif;
  margin-bottom: 7px;
  font-size: 17px;
  color: white;
}

.tv-card-img {
  box-shadow: 2px 2px 10px black;
  border-radius: 20px;
  position: relative;
  width: 210px;
}

.tv-card-image {
  border-radius: 20px;
}

.tv-card-image:hover {
  opacity: .5;
}

.view-series-btn:hover .tv-card-image {
  opacity: .5;
}

.tv-card-network {
  font-family: 'Raleway', sans-serif;
  font-size: 14px;
  margin: 0;
  color: #aea7b9;
}

