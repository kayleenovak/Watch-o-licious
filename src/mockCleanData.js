export const mockCleanCakeShows = [
  {
    id: 921,
    title: 'Cake Boss',
    runtime: 30,
    premiered: '2009-05-25',
    schedule: {
      'time': '21:00',
      'days': [
        'Monday'
      ]
    },
    rating: null,
    network: 'TLC',
    summary: 'Cake Boss is a popular docu-series following beloved baker Buddy Valastro, his larger-than-life family and burgeoning businesses.',
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/6/16129.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/6/16129.jpg"
  },
  {
    id: 27786,
    title: 'Extreme Cake Makers',
    runtime: 30,
    premiered: '2017-05-01',
    schedule: {
        time: "17:30",
        days: [
          "Monday",
          "Tuesday",
          "Wednesday",
          "Thursday",
          "Friday"
      ]
    },
    rating: null,
    network: 'Channel 4',
    summary: "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg"
  }
]

export const mockCleanBakeShows = [
    {
    id: 2950,
    title: 'The Great British Bake Off',
    runtime: 75,
    premiered: '2010-08-17',
    schedule: {
      time: "20:00",
      days: [
        "Tuesday"
      ]
    },
    rating: 8.7,
    network: 'Channel 4',
    summary: "Over 10 weeks, 12 of the best amateur bakers in Britain will whisk, knead, ice, beat and bake their way through classic British cakes, perfect patisserie, Italian delights, sticky caramel constructions and elaborate layered puddings. All 12 will be hoping to impress with their skill, creativity, knowledge and passion to clinch the Bake Off Crown. Each of the 30 new challenges have been carefully designed by judges Paul Hollywood and Prue Leith to reveal just who is a star baker. But it's not just a new experience for the bakers. Also joining the tent for the first time are Sandi Toksvig and Noel Fielding, who will be with the bakers every step of the way, ready with a pertinent pep talk, a helpful hand or just a sympathetic shoulder to cry on.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/124/311372.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/124/311372.jpg"
  },
    {
    id: 39695,
    title: 'The Great Kiwi Bake Off',
    runtime: 60,
    premiered: '2018-10-16',
    schedule: {
      time: "19:30",
      days: [
        "Tuesday"
      ]
    },
    rating: null,
    network: 'TVNZ 2',
    summary: "From biscuits to bread, pies to pavlovas, 12 amateur Kiwi bakers have their skills put to the test. Will they rise to the challenge?",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/174/435768.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/174/435768.jpg"
  }
]

export const mockCleanBakingShows = [
  {
    id: 1549,
    title: 'Kids Baking Championship',
    runtime: 60,
    premiered: '2015-02-02',
    schedule: {
      time: "21:00",
      days: [
          "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "Hosts and judges Duff Goldman and Valerie Bertinelli lead eight talented kid bakers through tasty challenges designed to find the most impressive and creative young baker on Kids Baking Championship. In each challenge, the kid contestants (ranging in age from 10 to 13) will be tested on their baking skills and originality, as they whip up delectable desserts including creative cakes, mouthwatering cookies and pastries making it seem as easy as pie. At stake, a sweet prize package that includes a complete set of Food Network bakeware products, a profile in Food Network Magazine, an original cake featured at Charm City Cakes and a $10,000 grand prize. Only one will take the cake to become the Kids Baking Champion!",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/88/220914.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/88/220914.jpg"
  },
  {
    id: 6407,
    title: 'Holiday Baking Championship',
    runtime: 60,
    premiered: '2014-11-09',
    schedule: {
      time: "21:00",
      days: [
        "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "On Holiday Baking Championship the search to find the greatest holiday baker begins as eight amateur bakers enter the kitchen, where they show off their family traditions and superb baking skills. To survive the challenges from week to week, they must prove their abilities in front of our tough-love judges: Duff Goldman, Nancy Fuller and Lorraine Pascale. Hosted by Bobby Deen, this baking championship is the sweetest way to celebrate this holiday season.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/24/61290.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/24/61290.jpg"
  }
]