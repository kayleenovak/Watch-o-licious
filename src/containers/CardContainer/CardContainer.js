import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import TvShowCard from '../TvShowCard/TvShowCard'
import './CardContainer.css'
import FavoriteCard from '../FavoriteCard/FavoriteCard.js'

export class CardContainer extends Component {

  shuffleTvShows = (tvShows) => {
    for (let i = tvShows.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [tvShows[i], tvShows[j]] = [tvShows[j], tvShows[i]];
    }
    return tvShows
  }

  splitLocation = (location) => {
    const splitLocation = location.split('/')
    return splitLocation[1]
  }
  
  render() {
    const { location } = this.props
    if(location === '/favorites' || location === '/watched' || location === '/watchlist') {
      const locationString = this.splitLocation(location)
      const trackedEpisodes = this.props.trackedEpisodes.filter(episode => {
        return episode.tracked[locationString] === true
      }).map(episode => {
        return <FavoriteCard />
      })
      return (
        <section className='tracked-episodes'>
          {trackedEpisodes}
        </section>
      )
    } else {
      const shuffledTvShows = this.shuffleTvShows(this.props.tvShows)
      const tvCards = shuffledTvShows.map(tvShow => {
        return <TvShowCard {...tvShow} />
      })
      return (
        <section className='card-container'>
          {tvCards}
        </section>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  tvShows: state.tvShows,
  trackedEpisodes: state.tracked
})

export default withRouter(connect(mapStateToProps)(CardContainer))