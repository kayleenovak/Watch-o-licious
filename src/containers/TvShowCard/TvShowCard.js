import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchTvShowInfo } from '../../thunks/fetchTvShowInfo.js'
import './TvShowCard.css'

export class TvShowCard extends Component {
  render() {
    const { title, network, image, fetchShowInfo } = this.props
    return (
      <article className='tv-show-card' onClick={() => fetchShowInfo(this.props.id)}>
        <img src={image} className='tv-card-img'/>
        <h3 className='tv-card-title'>{title}</h3>
        <p className='tv-card-network'>Network: {network}</p>
      </article>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchShowInfo: (id) => dispatch(fetchTvShowInfo(id))
})

export default connect(null, mapDispatchToProps)(TvShowCard)

