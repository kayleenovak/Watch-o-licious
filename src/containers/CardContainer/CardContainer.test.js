import React from 'react'
import { shallow } from 'enzyme'
import { isLoading, hasErrored} from '../../actions/index.js'
import { CardContainer, mapStateToProps} from './CardContainer.js'
import { mockCleanCakeShows } from '../../mockCleanData.js'
import TvShowCard from '../TvShowCard/TvShowCard.js'
import FavoriteCard from '../FavoriteCard/FavoriteCard.js'

describe('CardContainer', () => {
  let mockFavLocation
  let mockWatchedLocation
  let mockWatchListLocation
  let mockTrackedEpisodes
  let mockHomeLocation
  let mockFavoriteLocation

  beforeEach(() => {
    mockTrackedEpisodes = [{
      title: 'Cake Boss',
      tracked: {
        favorites: true,
        watched: true,
        watchlist: true
      }
    }]
    mockWatchedLocation = {
      pathname: '/watched'
    }
    mockFavoriteLocation = {
      pathname: '/favorites'
    }
    mockWatchListLocation = {
      pathname: '/watchlist'
    }
    mockHomeLocation = {
      pathname: '/'
    }
  })

  it('should match the snapshot', () => {
    const mockTvShows = []
    const wrapper = shallow(<CardContainer tvShows={ mockTvShows } location={ mockHomeLocation }/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render two TvShowCards if the location is home', () => {
    const wrapper = shallow(<CardContainer tvShows={ mockCleanCakeShows } location={ mockHomeLocation }/>)
    const spySplitLocaton = jest.spyOn(wrapper.instance(), 'displayTvShows')
    
    wrapper.instance().forceUpdate()

    expect(wrapper.find(TvShowCard).length).toBe(2)
    expect(spySplitLocaton).toHaveBeenCalled()
  })

  it('should render one Favorite Card if the location is favorites', () => {
    const wrapper = shallow(<CardContainer trackedEpisodes={ mockTrackedEpisodes} tvShows={ mockCleanCakeShows } location={ mockFavoriteLocation }/>)
    const spySplitLocaton = jest.spyOn(wrapper.instance(), 'splitLocation')
    const spyTrackedEpisodes = jest.spyOn(wrapper.instance(), 'findTrackedEpisodes')

    wrapper.instance().forceUpdate()

    expect(wrapper.find(FavoriteCard).length).toEqual(1)
  })

  it('should render one Favorite Card if the location is watched', () => {
    const wrapper = shallow(<CardContainer trackedEpisodes={ mockTrackedEpisodes} tvShows={ mockCleanCakeShows } location={ mockWatchedLocation }/>)
    const spySplitLocaton = jest.spyOn(wrapper.instance(), 'splitLocation')
    const spyTrackedEpisodes = jest.spyOn(wrapper.instance(), 'findTrackedEpisodes')

    wrapper.instance().forceUpdate()

    expect(wrapper.find(FavoriteCard).length).toEqual(1)
  })

  it('should render one Favorite Card if the location is watchlist', () => {
    const wrapper = shallow(<CardContainer trackedEpisodes={ mockTrackedEpisodes} tvShows={ mockCleanCakeShows } location={ mockWatchListLocation }/>)
    const spySplitLocaton = jest.spyOn(wrapper.instance(), 'splitLocation')
    const spyTrackedEpisodes = jest.spyOn(wrapper.instance(), 'findTrackedEpisodes')

    wrapper.instance().forceUpdate()

    expect(wrapper.find(FavoriteCard).length).toEqual(1)
  })

  it('splitLocation should return a location without /', () => {
    const expected = 'favorites'
    const wrapper = shallow(<CardContainer trackedEpisodes={ mockTrackedEpisodes} tvShows={ mockCleanCakeShows } location={ mockWatchListLocation }/>)
    const result = wrapper.instance().splitLocation('/favorites')

    expect(result).toEqual(expected)
  })

  describe('mapStateToProps', () => {
    const mockState = {
      tvShows: mockCleanCakeShows,
      isLoading,
      hasErrored
    }
    const expected = {
      tvShows: mockCleanCakeShows
    }

    const result = mapStateToProps(mockState)

    expect(result).toEqual(expected)
  })
})