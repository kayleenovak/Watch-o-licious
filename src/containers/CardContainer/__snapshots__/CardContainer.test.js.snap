// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    location={
      Object {
        "pathname": "/",
      }
    }
    tvShows={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
